charms/istio-pilot/metadata.yaml

```diff
@@ -17,5 +17,9 @@ provides:
     interface: ingress-auth
     schema: https://raw.githubusercontent.com/canonical/operator-schemas/master/ingress-auth.yaml
     versions: [v1]
+  gateway:
+    interface: gateway
```

Comment:
This is a very generic name, I don't think we can confidently create a gateway interface which could be listed in this [repository](https://github.com/canonical/charm-relation-interfaces). So lets be more specific with the name of the interface, e.g. `istio-gateway-name`